sdk/java-sdk-protobuf-testkit/src/main/java/kalix/javasdk/testkit/KalixTestKit.java

```diff
@@ -252,7 +274,15 @@ private void runProxy(Boolean useTestContainers, int port) {
       proxyContainer.addEnv("SERVICE_NAME", settings.serviceName);
       proxyContainer.addEnv("ACL_ENABLED", Boolean.toString(settings.aclEnabled));
       proxyContainer.addEnv("VIEW_FEATURES_ALL", Boolean.toString(settings.advancedViews));
-      proxyContainer.addEnv("JAVA_TOOL_OPTIONS", "-Dlogback.configurationFile=logback-dev-mode.xml");
+
+      List<String> javaOptions = new ArrayList<>();
+      javaOptions.add("-Dlogback.configurationFile=logback-dev-mode.xml");
+
+      settings.servicePortMappings.forEach((serviceName, hostPort) -> {
+        javaOptions.add("-Dkalix.dev-mode.service-port-mappings." + serviceName + "=" + hostPort);
+      });
```

Comment:
@efgpinto, this will conflict with your open PR. Make sure to rebase when this one gets merged. 

As the number of `-D` is growing, better if we use a list and then join it.